### PR TITLE
Use Node names for vertices in dotty DAG dump

### PR DESCRIPTION
**Pull request by nickgg**
Summary: Now that all nodes in a Function, and all Storage names are unique, we can use the node name as the vertex name in the DAG output rather than a random string of numbers. This has the advantage of making the same node from different dot files use the name Vertex name, which i need for a _\~secret project\~_.

Documentation: N/A

Test Plan: ran unit tests, dumped a DAG and visualized it.

### CODE REVIEW

File: lib/Graph/Graph.cpp [hunk 1]
```diff
@@ -276,14 +276,18 @@ class AbstractDottyPrinter {
...
+  void dumpNode(Node *N, bool uniqueNames) {
     if (!N) {
       return;
     }
     std::ostringstream os;
-    // Print a node descriptor that looks like this:
-    // vNNNN [ shape = "record" label = "{...}" ];
-    os << uniqueVertexName(N) << "[\n";
+    if (uniqueNames) {
+      // Print a node descriptor that looks like this:
```
Comment (jfix71) [inline lib/Graph/Graph.cpp]: Think this comment should be bumped back above the if block here.

File: lib/Graph/Graph.cpp [hunk 2]
```diff
@@ -381,7 +385,7 @@ class ModuleDottyPrinter : public AbstractDottyPrinter {
 public:
   void visitModule(Module *M) {
     for (auto N : M->getConstants()) {
-      dumpNode(N);
+      dumpNode(N, true);
```
Comment (jfix71) [inline lib/Graph/Graph.cpp]: Why do we need to still use the random string of numbers in this case?

Comment (nickgg) [reply lib/Graph/Graph.cpp]: Because a module may have multiple functions with the same nodes, however they should not be the same vertices in the dot output.

File: lib/Graph/Graph.cpp (hunk 1 above)
Comment (nickgg) [reply lib/Graph/Graph.cpp]: The node descriptor doesn't look like that though, would you like me to add a comment with how the other node descriptor looks?

Comment (jfix71) [reply lib/Graph/Graph.cpp]: Oh -- well I think the original comment was referring to everything being printed, not just the initial vertex name. NBD.

File: lib/Graph/Graph.cpp (hunk 2 above)
Comment (jfix71) [reply lib/Graph/Graph.cpp]: Isn't this handling Constants though? Clearly I'm confused 🤷‍♂ maybe add a comment explaining what's going on?

Comment (nickgg) [reply lib/Graph/Graph.cpp]: It's a bit intricate, yeah. This block is handling constants, and is setting the Vertex name of the constant so we can add edges to it later. In this path where we're dumping the Module we use a version of the Function that does node uniquing since it might have multiple vertices for the same Node name. Since we're doing uniquing all vertices need to use the same uniquing rule in order for us to create the right edges.

TLDR: use node names for single function dumping, use uniqued names for Module dumping.

File: lib/Graph/Graph.cpp (hunk 1 above)
Comment (nickgg) [reply lib/Graph/Graph.cpp]: I'll clean this up.
